### PR TITLE
chore(logging): add time to graphql log

### PR DESCRIPTION
**Pull request by Avantol13**
### Improvements
- Improved logging for GraphQL queries (include timing)

### CODE REVIEW

File: src/server/es/index.js [hunk 1]
```diff
@@ -53,14 +53,20 @@ class ES {
...
+
+    var start = new Date().getTime();
+    var searchResult = this.client.search({
       index: esIndex,
       body: validatedQueryBody,
     }).then((resp) => resp.body, (err) => {
       log.error(`[ES.query] error during querying: ${err.message}`);
       throw new Error(err.message);
     });
+    var end = new Date().getTime();
```
Comment (k-burt-uch) [inline src/server/es/index.js:65]: this.client.search is async so this end time won't reflect when the request ends, just when it was sent. 

you'll want to add a call to finally after the .then and add this and the code below.

**Review by k-burt-uch — CHANGES_REQUESTED** [review]

File: src/server/es/index.js (hunk 1 above)
Comment (Avantol13) [reply src/server/es/index.js:65]: Good catch, I'll fix it. Is it obvious I don't with Javascript very often? :laughing:

Comment (k-burt-uch) [reply src/server/es/index.js:65]: something like this 

```
var searchResult = this.client.search({
  index: esIndex,
  body: validatedQueryBody,
}).then((resp) => resp.body, (err) => {
  log.error(`[ES.query] error during querying: ${err.message}`);
  throw new Error(err.message);
}).finally(() => {
   var end = new Date().getTime();
   var durationInMS = end - start;

   log.info('[ES.query] DurationInMS:' + durationInMS + '. index, type, query body: ', esIndex, esType, JSON.stringify(validatedQueryBody));
   return searchResult;
})
```

and no worries i almost commented on the log.info asking you to use f strings before realizing this was javascript

File: src/server/es/index.js [hunk 2]
```diff
@@ -53,14 +53,21 @@ class ES {
...
     }).then((resp) => resp.body, (err) => {
       log.error(`[ES.query] error during querying: ${err.message}`);
       throw new Error(err.message);
-    });
+    }).finally(() => {
+       var end = new Date().getTime();
+       var durationInMS = end - start;
+
+       log.info('[ES.query] DurationInMS:' + durationInMS + '. index, type, query body: ', esIndex, esType, JSON.stringify(validatedQueryBody));
+       return searchResult;
```
Comment (k-burt-uch) [inline src/server/es/index.js:69]: Don't return from the finally - you can actually get rid of searchResult assignment and `return this.client.search(...)` like it was before. This function needs to return the Promise for the async call so callers of the functions can do their own async response handling.

**Review by k-burt-uch — CHANGES_REQUESTED** [review]

File: src/server/es/index.js (hunk 2 above)
Comment (Avantol13) [reply src/server/es/index.js:69]: :+1: I'll push and then I'm going to test in my dev env

**Review by AlbertSnows — DISMISSED** [review]

File: src/server/es/index.js [hunk 3]
```diff
@@ -53,14 +53,20 @@ class ES {
...
     }).then((resp) => resp.body, (err) => {
       log.error(`[ES.query] error during querying: ${err.message}`);
       throw new Error(err.message);
-    });
+    }).finally(() => {
+       var end = Date.now();
+       var durationInMS = end - start;
+
+       log.info('[ES.query] DurationInMS:' + durationInMS + '. index, type, query body: ', esIndex, esType, JSON.stringify(validatedQueryBody));
+    })
```
Comment (k-burt-uch) [inline src/server/es/index.js:69]:
```suggestion
    });
```

**Review by k-burt-uch — APPROVED** [review]

File: src/server/es/index.js (hunk 3 above)
Comment (Avantol13) [reply src/server/es/index.js]: Geez. I've been working with too much Python. Thanks

How many engineers does it take to add a log :sweat_smile:

**Pull request merged**